Sources/Swift/Persistence/SentryScopePersistentStore.swift

```diff
@@ -220,3 +265,78 @@ extension SentryScopePersistentStore {
         }
     }
 }
+
+// MARK: - Level
+extension SentryScopePersistentStore {
+    private func encode(level: NSNumber) -> Data? {
+        let rawValue = level.uintValue
+        return Data("\(rawValue)".utf8)
+    }
+    
+    private func decodeLevel(from data: Data) -> SentryLevel? {
+        if let stringValue = String(data: data, encoding: .utf8),
+           let intValue = UInt(stringValue),
+            let level = SentryLevel(rawValue: intValue) {
+            return level
+        } else {
+            return nil
+        }
+    }
+}
+
+// MARK: - Extras
+extension SentryScopePersistentStore {
```

Comment:
`m`: I like the structure using extension, but this file is now at 342 lines, so how about having files like `SentryScopePersistentStore+Extras.swift` where the encoding/decoding for each field is?